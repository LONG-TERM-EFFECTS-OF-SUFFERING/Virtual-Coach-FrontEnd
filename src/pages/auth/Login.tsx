import { useState, useEffect } from "react"
import { Link, redirect } from "react-router-dom"
import { connect } from "react-redux"
import { PiPasswordLight, PiUserLight } from 'react-icons/pi';
import { AiOutlineEye, AiOutlineEyeInvisible } from 'react-icons/ai';
import {loginType} from '../../interfaces/auth'
import { login } from '../../actions/auth';


const Login: React.FC<loginType>= ({login}) => {


  const [formData, setFormData] = useState({
    email: "",
    password: "",
  })

  const [showPassword, setShowPassword] = useState(false)

  const handleTogglePassword = () => {
    setShowPassword(!showPassword);
  };
  

  const onChange = (e:any) => setFormData({ ...formData, [e.target.name]: e.target.value });

  const { email, password } = formData;

  const onSubmit =  (e:any) => {
    e.preventDefault()
    login(email, password);
  }
  return (
    <div className="bg-white p-8 w-full md:w-96 rounded-xl">
      
      <h1 className="text-2xl font-bold mb-5 text-center">Iniciar Sesión</h1>
      <form className="flex flex-col gap-4" onSubmit={e => onSubmit(e)}>
        <div className="relative">
          <PiUserLight className={"absolute top-2 left-2"}/>
          <input type="email" className="bg-gray-100 border w-full outline-none px-4 pl-7 rounded-lg py-1"  placeholder="email" onChange={e => onChange(e)} name="email" value={email}/>
        </div>
        <div className="relative">
          <PiPasswordLight className="absolute top-2 left-2"/>
          <input type={showPassword ? "text" : "password" } className="bg-gray-100 border w-full outline-none px-8  rounded-lg py-1"  placeholder="contraseña" onChange={e => onChange(e)} name="password" value={password}/>
          {showPassword ? (
            <AiOutlineEye className="absolute top-2 right-2 hover:cursor-pointer" onClick={handleTogglePassword}/>
          ) : (
            <AiOutlineEyeInvisible className="absolute top-2 right-2 hover:cursor-pointer" onClick={handleTogglePassword}/>
          )}

        </div>
        <div>
          <button className="w-full mt-5 bg-sky-500 hover:bg-indigo-500 hover:scale-105  transition-all text-white rounded-md py-1 font-bold" type='submit'>
            Login
          </button>
        </div>
      </form>
    </div>  

  )
}

// const mapStateToProps = (state: any) => ({
//   isAuthenticated: state.auth.isAuthenticated,
// })


export default connect(null, {login})(Login)