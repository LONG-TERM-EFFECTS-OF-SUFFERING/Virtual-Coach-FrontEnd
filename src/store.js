import {createStore, applyMiddleware} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReduxer from './reducers';

const initailState = {};
const middleware = [thunk];

const store = createStore(
    rootReduxer,
    initailState,
    composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
