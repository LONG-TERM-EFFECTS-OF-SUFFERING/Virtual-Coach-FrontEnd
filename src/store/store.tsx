import { configureStore } from '@reduxjs/toolkit';
import { persistReducer, persistStore } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import user from './slices/user/user';
import thunk from 'redux-thunk';


const persistConfig:any = {
    key: 'user',
    storage,
}

const persistedReducer = persistReducer(persistConfig, user)

export const store = configureStore({
    reducer: persistedReducer,
    middleware: [thunk]
})
export const persistor = persistStore(store)

export type RootState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch
