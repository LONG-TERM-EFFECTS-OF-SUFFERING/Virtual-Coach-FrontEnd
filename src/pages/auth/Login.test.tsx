import { test, expect, beforeEach, vi } from 'vitest';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import axios from 'axios';
import { store } from '../../store/store';
import Login from './Login';

const component = (
    <Provider store={store}>
        <Router>
            <Login />
        </Router>
    </Provider>
)

beforeEach(() => {
    vi.resetAllMocks();
    cleanup()
})

test('Login renders correctly', async () => {
    const { container } = render(component)
    expect(container).toMatchSnapshot()
})

test('Error message is shown when user_login fails', async () => {
    const data = { message: 'Error'}
    vi.spyOn(axios, 'post').mockImplementationOnce(() => {throw {response: {data}}})

    render(component)
    const login_button = screen.getByText('Login')
    fireEvent.click(login_button)

    const alert = await screen.findByText(data.message)
    expect(alert).toBeDefined()
})

test('Error message is not shown when user_login succeeds', async () => {
    const data = { message: 'Success'}
    vi.spyOn(axios, 'post').mockImplementationOnce(() => {
        return Promise.resolve({data})
    })

    render(component)
    const login_button = screen.getByText('Login')
    fireEvent.click(login_button)

    const alert = await screen.queryByText(data.message)
    expect(alert).toBeNull()
})