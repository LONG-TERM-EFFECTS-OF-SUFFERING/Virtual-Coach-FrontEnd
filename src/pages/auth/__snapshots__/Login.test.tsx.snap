// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login renders correctly 1`] = `
<div>
  <div
    class="bg-white p-8 w-full md:w-96 rounded-xl"
  >
    <div
      class="mb-3"
    />
    <h1
      class="text-2xl font-bold mb-5 text-center"
    >
      Iniciar Sesión
    </h1>
    <form
      class="flex flex-col gap-4"
    >
      <div
        class="relative"
      >
        <svg
          class="absolute top-2 left-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M229.19,213c-15.81-27.32-40.63-46.49-69.47-54.62a70,70,0,1,0-63.44,0C67.44,166.5,42.62,185.67,26.81,213a6,6,0,1,0,10.38,6C56.4,185.81,90.34,166,128,166s71.6,19.81,90.81,53a6,6,0,1,0,10.38-6ZM70,96a58,58,0,1,1,58,58A58.07,58.07,0,0,1,70,96Z"
          />
        </svg>
        <input
          class="bg-gray-100 border w-full outline-none px-4 pl-7 rounded-lg py-1"
          name="email"
          placeholder="email"
          type="email"
          value=""
        />
      </div>
      <div
        class="relative"
      >
        <svg
          class="absolute top-2 left-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M46,56V200a6,6,0,0,1-12,0V56a6,6,0,0,1,12,0Zm86.58,56.41L110,119.74V96a6,6,0,0,0-12,0v23.74l-22.58-7.33a6,6,0,1,0-3.71,11.41l22.58,7.33-14,19.21a6,6,0,1,0,9.7,7.06l14-19.21,14,19.21a6,6,0,0,0,9.7-7.06l-14-19.21,22.58-7.33a6,6,0,1,0-3.71-11.41Zm103.56,3.85a6,6,0,0,0-7.56-3.85L206,119.74V96a6,6,0,0,0-12,0v23.74l-22.58-7.33a6,6,0,1,0-3.71,11.41l22.58,7.33-13.95,19.21a6,6,0,1,0,9.7,7.06l14-19.21,14,19.21a6,6,0,0,0,9.7-7.06l-13.95-19.21,22.58-7.33A6,6,0,0,0,236.14,116.26Z"
          />
        </svg>
        <input
          class="bg-gray-100 border w-full outline-none px-8  rounded-lg py-1"
          name="password"
          placeholder="contraseña"
          type="password"
          value=""
        />
        <svg
          class="absolute top-2 right-2 hover:cursor-pointer"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
          />
          <path
            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
          />
        </svg>
      </div>
      <div>
        <button
          class="w-full mt-5 bg-sky-500 hover:bg-indigo-500 hover:scale-105  transition-all text-white rounded-md py-1 font-bold"
          type="submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;
